modules/markup/markdown/transform_blockquote.go

```diff
@@ -87,12 +104,38 @@ func (g *ASTTransformer) transformBlockquote(v *ast.Blockquote, reader text.Read
 	emphasis.AppendChild(emphasis, attentionAstString)
 	attentionParagraph.AppendChild(attentionParagraph, NewAttention(attentionType))
 	attentionParagraph.AppendChild(attentionParagraph, emphasis)
-	firstParagraph.Parent().InsertBefore(firstParagraph.Parent(), firstParagraph, attentionParagraph)
-	firstParagraph.RemoveChild(firstParagraph, node1)
-	firstParagraph.RemoveChild(firstParagraph, node2)
-	firstParagraph.RemoveChild(firstParagraph, node3)
-	if firstParagraph.ChildCount() == 0 {
-		firstParagraph.Parent().RemoveChild(firstParagraph.Parent(), firstParagraph)
+	return attentionParagraph
+}
+
+func (g *ASTTransformer) transformBlockquote(v *ast.Blockquote, reader text.Reader) (ast.WalkStatus, error) {
+	// We only want attention blockquotes when the AST looks like:
+	// > Text("[") Text("!TYPE") Text("]")
+	//
+	// or, in case of a math block: \[!TYPE\]
+
+	firstChild := v.FirstChild()
+	var attentionType string
+
+	// grab attention type from markdown source
+	if paragraph, ok := firstChild.(*ast.Paragraph); ok {
+		attentionType = popAttentionTypeFromParagraph(g, paragraph, reader)
+	} else {
+		mathBlock, ok := firstChild.(*math.Block)
```

Comment:
Maybe in the future we should use an "Extension" like "math" , instead of using "transform", then no need to play the type-casting trick with "math.Block" (just an idea, not a change suggestion 😁 )